### ISSUE TITLE
Implement stats

### ISSUE THREAD

**Issue by curquiza** [meilisearch/milli#113]
Implement stats (general and of an index) so that transplant can have an iso behavior with the current version of MeiliSearch

More information on [the MeiliSearch /stats reference documentation](https://docs.meilisearch.com/reference/api/stats.html#stats).

**Comment by Kerollmops** [comment]
There are two ways of implementing the `fieldsDistribution` part of the stats route, the simple and the little more complex way.

## The simple way

Don't keep any state and recompute the fields distribution every time a user request the stats. It is a simple as iterating over the index document database, [reading the obkvs](https://docs.rs/obkv/) and increasing a counter for each one of the fields present. This approach can be slow when the number of documents becomes important, but it is a valid implementation.

```rust
impl Index {
    fn stats(&self, &rtxn: &RoTxn) -> anyhow::Result<Stats> {

        for result in self.documents.iter(rtxn)? {
            let (_id, obkv) = result?;
            // fetch and increase a counter for every field in this document (obkv)
        }

        unimplemented!()
    }
}
```

## The little more complex way

This implementation version is as simple as computing the `fieldsDistribution` stats and storing them into the Index itself after every update, an even better solution would be to do so only for any updates that modify the documents.

This solution could be implemented afterward because it needs more carefulness on where and when we should reset the stored stats.

**Comment by Kerollmops** [comment]
Thank you @shekhirin for taking this issue, you are a hell of a contributor 🎉

It could, indeed, be a good idea to introduce a `fields_distribution` method on `Index` that would return an `HashMap` with the number of documents that have the specified fields. I am not sure if it would be better to return `FieldId` as keys or directly the field names (converted by using the `Index::fields_ids_map` method).

For now, we could expose a `field_ids_distribution` method on `Index` that would return an `HashMap` with `FieldId`s as keys and an `u64` for the count, @MarinPostma will convert it itself until he finds it easier than we do it on our side. In this case, we will patch the method. Anyway, that is not exposed to anyone but the MeiliSearch team for now, so we can break things :)

Thank you very much, again!

**Comment by curquiza** [comment]
@Kerollmops can this issue be closed? I mean, does milli return enough information to let transplant implement the `stats/` and `/indexes/:uid/stats` routes?

**Comment by shekhirin** [comment]
Another thing you may want to implement in this repo is calculating the db size. 

I see that previously it was done by traversing the whole db directory
```rust
    let database_size = WalkDir::new(&data.db_path)
        .into_iter()
        .filter_map(|entry| entry.ok())
        .filter_map(|entry| entry.metadata().ok())
        .filter(|metadata| metadata.is_file())
        .fold(0, |acc, m| acc + m.len());
```

However I'm not quite sure it needs to be there and not in the transplant repo.

**Comment by Kerollmops** [comment]
You are right @shekhirin, it should be in the transplant repo as it is the HTTP part that knows about the update store and different indexes databases.

**Comment by Kerollmops** [comment]
I think we can close this issue as @shekhirin have implemented #135 and #136, thank you again for your hard work 😃
